### PR TITLE
Сounterparties

### PR DESCRIPTION
**Pull request by Slezaintima**

### CODE REVIEW

File: App.Common/Repositories/ICustomersRepository.cs [hunk 1]
```diff
@@ -1,6 +1,13 @@
-﻿namespace App.Repositories
+﻿using System.Collections.Generic;
+
+namespace App.Repositories
 {
     public interface ICustomersRepository
     {
+        int Size { get; }
```
Comment (Slezaintima) [inline App.Common/Repositories/ICustomersRepository.cs]: This is incapsulation violation

File: App.Common/Repositories/ICustomersRepository.cs [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+        IEnumerable<string> GetCustomers();
+        string GetCustomer(int id);
```
Comment (Slezaintima) [inline App.Common/Repositories/ICustomersRepository.cs]: you mst introduce class Customer in App.Common.Models

File: App.Customers/Controllers/CustomerController.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace App.Customers.Controllers
+{
+    [Route("api/customers")]
+    [ApiController]
+    public class CustomerController:ControllerBase
+    {
+        readonly ILogger<CustomerController> _logger;
```
Comment (Slezaintima) [inline App.Customers/Controllers/CustomerController.cs]: remove the ogger, it is part of next laboratory work

File: App.Customers/Controllers/CustomerController.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        public ActionResult<string> GetById( int id)
+        {
+            _logger.LogInformation("NOTHING");
+            var serviceCallResult = _customersManager.GetCustomer(id);
+            return serviceCallResult;
+        }
+
+
+        [HttpPost]
+        public ActionResult Add(string customer)
```
Comment (Slezaintima) [inline App.Customers/Controllers/CustomerController.cs]: add Cstomer clas sing

File: App.Customers/Controllers/CustomerController.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            _logger.LogInformation("NOTHING");
+            _customersManager.Add(customer);
+            return Ok();
+        }
+
+        [HttpPut]
+        public ActionResult Update(string oldCustomer, string newCustomer)
```
Comment (Slezaintima) [inline App.Customers/Controllers/CustomerController.cs]: should use class

File: App.Customers/CustomerManager.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+using System.Text;
+using App.Configuration;
+using App.Customers;
+using App.Repositories;
+
+namespace App.Customers
+{
+    public interface ICustomersManager
+    {
+        IEnumerable<string> GetCustomers();
```
Comment (Slezaintima) [inline App.Customers/CustomerManager.cs]: use class Customrt

File: App.Customers/CustomerManager.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        string GetCustomer(int id);
+        void Add(string customers);
+        bool Update(string oldCustomer, string newCustomer);
```
Comment (Slezaintima) [inline App.Customers/CustomerManager.cs]: change implementation ofupdate method

File: App.Customers/CustomerManager.cs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    public class CustomerManager:ICustomersManager//,ITransientDependency
```
Comment (Slezaintima) [inline App.Customers/CustomerManager.cs]: remove comment

File: App.Customers/Repositories/CustomersRepository.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using App.Configuration;
+using App.Repositories;
+
+namespace App.Customers.Repositories
+{
+    public class CustomersRepository : ICustomersRepository//, ITransientDependency
```
Comment (Slezaintima) [inline App.Customers/Repositories/CustomersRepository.cs]: remove comment

File: App.Customers/Repositories/CustomersRepository.cs [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    {
+
+        static List<string> customersList = new List<string>();
```
Comment (Slezaintima) [inline App.Customers/Repositories/CustomersRepository.cs]: init list of customers in static constructor

File: App.Customers/Repositories/CustomersRepository.cs [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        public int Size{ get => customersList.Count; }
+        public IEnumerable<string> GetCustomers()
+        {
+            customersList.Add("dasfsds");
```
Comment (Slezaintima) [inline App.Customers/Repositories/CustomersRepository.cs]: this method shoud olny return list of cstomers

File: App.Web/App.Web.csproj [hunk 12]
```diff
@@ -1,8 +1,11 @@
-<Project Sdk="Microsoft.NET.Sdk.Web">
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
     <TargetFramework>netcoreapp2.2</TargetFramework>
     <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
+    <StartupObject></StartupObject>
```
Comment (Slezaintima) [inline App.Web/App.Web.csproj]: remove extra code

File: App.Web/App.Web.csproj [hunk 13]
```diff
@@ -31,7 +34,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\App.Common\App.Common.csproj" />
-    <ProjectReference Include="..\App.TestModule\App.Example.csproj" />
```
Comment (Slezaintima) [inline App.Web/App.Web.csproj:34]: restore test modue

File: App.Web/Modules.cs [hunk 14]
```diff
@@ -5,7 +5,7 @@ namespace App.Web
     /// <summary>
     /// IMPORTANT ! In order to use classes and endpoints, defined in your own module, it should be referenced here as it shown
     /// </summary>
-    [ModuleUsing(typeof(Example.ExampleModule))] // < ---- Example of module registration
```
Comment (Slezaintima) [inline App.Web/Modules.cs:8]: restore reference to test modules

File: App.Web/App.Web.csproj [hunk 15]
```diff
@@ -1,8 +1,10 @@
-<Project Sdk="Microsoft.NET.Sdk.Web">
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
     <TargetFramework>netcoreapp2.2</TargetFramework>
     <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
+    <ApplicationIcon />
```
Comment (Slezaintima) [inline App.Web/App.Web.csproj:6]: revert this

**Review by ghost — APPROVED** [review]

**Pull request merged**